MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/location/LocationComponentTest.kt

```diff
@@ -386,6 +386,24 @@ class LocationComponentTest {
     verify(compassEngine, never()).addCompassListener(any(CompassListener::class.java))
   }
 
+  @Test
+  fun compass_notAddListenerWhenLayerNotReady() {
+    locationComponent.activateLocationComponent(context, mock(Style::class.java), locationEngine, locationEngineRequest, locationComponentOptions)
+    locationComponent.isLocationComponentEnabled = true
+    locationComponent.onStart()
+    `when`(locationLayerController.isConsumingCompass).thenReturn(true)
+
+    locationComponent.onStartLoadingMap()
+    // Layer should be disabled at this point
+    locationComponent.setCameraMode(CameraMode.TRACKING)
```

Comment:
NIT: It'd be better to use a mode that actually consumes the compass, like `setRenderMode(RenderMode.COMPASS)`. It's not impacting the test because of the `locationLayerController` mock above but it would be good to correct.

Reply:
Adjusted accordingly.

Reply:
Still show up the same for me :upside_down_face: Did you push all of the local changes?

Reply:
Sorry I misunderstood your first comment. I was trying to reproduce the issue you mentioned in https://github.com/mapbox/mapbox-gl-native/issues/14889#issuecomment-541128492 , where the compass listeners are re-enabled when the camera is changed before the style is loaded. Firstly I call `locationComponent.onStartLoadingMap()`, it will change the `isLayerReady` flag to false, then I change the camera mode, internally, it would call `updateCompassListenerState()`, and I will verify that the compass listeners are not added because the Layer is not ready.

Reply:
All is fine, just that `CameraMode.TRACKING` doesn't actually use compass data. That's taken into account in the camera implementation itself, but if it were to change, this test might not hit the target anymore. Anyway, we can address later if needed.

Reply:
sorry @LukasPaczos I just got it, you meant change to something like `locationComponent.setCameraMode(CameraMode.TRACKING_COMPASS)` not `setRenderMode(RenderMode.COMPASS)`, will refactor to it.